src/common/buffer.cc

```diff
@@ -536,6 +537,7 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
 	      << bendl;
 	throw error_code(r);
       }
+      auto sg = make_scope_guard([&] { close_pipe(tmpfd); });	  
```

Comment:
you change should work. but i'd suggest not capture `this` if not necessary. maybe we can mark `close_pipe()` a static method, like:
```diff
diff --git a/src/common/buffer.cc b/src/common/buffer.cc
index 7ca3a66d29..5113e8e5d8 100644
--- a/src/common/buffer.cc
+++ b/src/common/buffer.cc
@@ -515,7 +515,7 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
       return 0;
     }

-    void close_pipe(int *fds) {
+    static void close_pipe(const int *fds) {
       if (fds[0] >= 0)
        VOID_TEMP_FAILURE_RETRY(::close(fds[0]));
       if (fds[1] >= 0)
@@ -537,7 +537,7 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
              << bendl;
        throw error_code(r);
       }
-      auto sg = make_scope_guard([&] { close_pipe(tmpfd); });
+      auto sg = make_scope_guard([=] { close_pipe(tmpfd); });
       r = set_nonblocking(tmpfd);
       if (r < 0) {
        bdout << "raw_pipe: error setting nonblocking flag on temp pipe: "
```